inventory_management_system_api/schemas/setting.py

```diff
@@ -0,0 +1,65 @@
+"""
+Module for defining the API schema models for representing settings.
+"""
+
+from pydantic import BaseModel, Field, conlist, field_validator
+
+from inventory_management_system_api.schemas.usage_status import UsageStatusSchema
+
+
+class SparesDefinitionPutUsageStatusSchema(BaseModel):
+    """
+    Schema model for a usage status within a spares definition update request.
+    """
+
+    id: str = Field(description="The ID of the usage status.")
+
+
+class SparesDefinitionUsageStatusSchema(SparesDefinitionPutUsageStatusSchema):
+    """
+    Schema model for a usage status within a spares definition.
+    """
+
+    id: str = Field(description="The ID of the usage status.")
+
+
+class SparesDefinitionPutSchema(BaseModel):
+    """
+    Schema model for a spares definition update request.
+    """
+
+    usage_statuses: conlist(SparesDefinitionPutUsageStatusSchema, min_length=1) = Field(
+        description="Usage statuses that classify items as a spare."
+    )
+
+    @field_validator("usage_statuses")
+    @classmethod
+    def validate_usage_statuses(
+        cls, usage_statuses: list[SparesDefinitionPutUsageStatusSchema]
+    ) -> list[SparesDefinitionPutUsageStatusSchema]:
```

Comment:
The `usage_statuses` param is type casted to `SparesDefinitionPutUsageStatusSchema`, but when it's reused in the `SparesDefinitionSchema`, its type for `usage_statuses` is `UsageStatusSchema`. There doesn't seem to be any errors caused by this, since they both have the same id field, but the param's type could be changed to a union of the 2 schemas.

Reply:
True, good spot! I think looking over the others, this is the only case where it occurs, I think its best to avoid having the validator on the get though as we only really wanted it on the post in the first place. In this case the simplest solution is to just inherit from `BaseModel` instead of `SparesDefinitionPutSchema` as its only overwriting the one field anyway.